### PR TITLE
Adds smart handling of response objects

### PR DESCRIPTION
**Pull request by seemethere**
Returning dicts in handlers will now convert them to json and returning anything other than an HTTPResponse will return its `str` representation. 

In response to #50 

# Old
```python
from sanic import Sanic
from sanic.response import json
app = Sanic('test_json')

@app.route('/')
async def handler(request):
    return json({"test": True})
```

# New
```python
from sanic import Sanic
app = Sanic('test_json')

@app.route('/')
async def handler(request):
    return {"test": True}
```

### CODE REVIEW

File: sanic/server.py [hunk 1]
```diff
@@ -121,9 +122,20 @@ def on_message_complete(self):
     # -------------------------------------------- #
 
     def write_response(self, response):
+        """Attempts to write the resposne to the transport
+
+        This will attempt to smartly handle any type of python object with
```
Comment (rzwiefel) [inline sanic/server.py]: Might sound a little better if you say something like ...`attempt to intelligently handle`...

**Comment by rzwiefel** [comment]
Sorry for the pedantic little comments. They just caught my eye 🙂

**Comment by seemethere** [comment]
@rzwiefel No problem! I agree with the comments!

**Comment by channelcat** [comment]
I really like this.  If it doesn't slow the benchmarks I'd like to merge it :).  A few questions though:
1. How do we want to handle a response of None?
2. Should a string response be html or text by default?
3. IMO response type conversion should be the concern of the framework and in sanic.py.  If the http server were ever distributed on its own, this would impose the type conversion on other frameworks.  What are your thoughts?

**Comment by seemethere** [comment]
@channelcat would you be talking about here? 
https://github.com/channelcat/sanic/blob/master/sanic/sanic.py#L140

**Comment by channelcat** [comment]
@seemethere yep!  It did slow things down a little bit so I'd like to keep discussing - gonna post in https://github.com/channelcat/sanic/issues/50

**Comment by seemethere** [comment]
@channelcat 

I'm not seeing a remarkable slow down when I run `wrk`

I'll close this pull request for now while we discuss the feature in #50.

Results:
```shell
bash-3.2$ wrk -c 100 http://0.0.0.0:8000  # With smart handling
Running 10s test @ http://0.0.0.0:8000
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    34.79ms   30.90ms 282.59ms   93.58%
    Req/Sec     1.28k   753.04     4.29k    79.84%
  16434 requests in 10.10s, 1.85MB read
Requests/sec:   1626.61
Transfer/sec:    187.44KB
bash-3.2$ wrk -c 100 http://0.0.0.0:8000  # Without smart handling
Running 10s test @ http://0.0.0.0:8000
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    33.09ms   31.34ms 301.59ms   94.33%
    Req/Sec     1.29k   724.03     4.08k    80.16%
  16444 requests in 10.04s, 1.85MB read
Requests/sec:   1637.13
Transfer/sec:    188.65KB
```

**Pull request closed without merging**